spec/mailers/job_mailer_spec.rb

```diff
@@ -19,11 +28,50 @@
   end
 
   describe 'body' do
-    before { job_run.state = 'discovery_report_complete' }
+    context 'for a discovery report email' do
+      before { job_run.state = 'discovery_report_complete' }
+
+      it 'renders the body' do
+        expect(job_notification.body.encoded).to include("Your Discovery report job ##{job_run.id} finished with status 'Discovery report completed'")
+          .and include("http://localhost:3000/job_runs/#{job_run.id}")
+      end
+    end
+
+    context 'for a preassembly job report email' do
+      before do
+        job_run.job_type = 1
```

Comment:
iirc, i think you can also set `job_type` to the string value (`'preassembly'`), and activerecord magic will do the translation for you, which might make this slightly more readable.